src/Microsoft.AspNet.Hosting/Internal/HostingLoggerExtensions.cs

```diff
@@ -108,29 +133,66 @@ public override string ToString()
                 return _cachedToString;
             }
 
-            public IEnumerable<KeyValuePair<string, object>> GetValues()
+            public IEnumerator<KeyValuePair<string, object>> GetEnumerator()
             {
-                if (_cachedGetValues == null)
+                return new List<KeyValuePair<string, object>>
                 {
-                    _cachedGetValues = new[]
-                    {
-                        new KeyValuePair<string, object>("RequestId", _httpContext.TraceIdentifier),
-                        new KeyValuePair<string, object>("RequestPath", _httpContext.Request.Path.ToString()),
-                    };
-                }
+                    new KeyValuePair<string, object>("RequestId", _httpContext.TraceIdentifier),
+                    new KeyValuePair<string, object>("RequestPath", _httpContext.Request.Path.ToString())
+                }.GetEnumerator();
+            }
 
-                return _cachedGetValues;
+            IEnumerator IEnumerable.GetEnumerator()
+            {
+                return GetEnumerator();
             }
         }
 
-        private class HostingRequestStarting : ILogValues
+        private class HostingRequestStarting : IReadOnlyList<KeyValuePair<string, object>>
```

Comment:
Is there a reason to implement `IReadOnlyList` if we provide a message formatter already?